sdk/core/src/tasks/sboms/github/sync.rs

```diff
@@ -223,71 +252,120 @@ impl SyncTask {
         &self,
         repo: &mut Repo,
         _task: &mut Task,
-    ) -> Result<Option<String>, Error> {
-        let url: &String = match &repo.html_url {
-            Some(url) => url,
-            None => return Err(Error::GitHub("==> No URL for Repository".to_string())),
-        };
+        cataloger_to_build_file_map: &HashMap<String, Vec<String>>,
+    ) -> Result<Option<Vec<String>>, Error> {
+        let (url, full_name) = self.get_state_values(repo)?;
 
-        let full_name: &String = match &repo.full_name {
-            Some(full_name) => full_name,
-            None => return Err(Error::GitHub("==> Repo is missing full_name".to_string())),
-        };
-
-        let last_hash: &String = &repo.last_hash;
-
-        println!(
-            "==> Looking in Mongo for the document with repo url: {}",
-            url
-        );
-
-        let mut document = match self.mongo_svc.find(url).await {
-            Ok(option) => match option {
-                Some(document) => {
-                    println!("==> Got a Document From Mongo!");
-                    document
-                }
-                None => {
-                    println!("==> No document exists in mongo with the id: {}", url);
-                    self.mongo_svc
-                        .create_document(url.clone(), String::from(""))
-                        .await
-                        .map_err(|e| Error::GitHub(e.to_string()))?
-                }
-            },
+        // The url is the id of the document
+        let mut document = match self.find_document(url.as_str()).await {
+            Ok(document) => document,
             Err(err) => {
                 return Err(Error::GitHub(format!(
-                    "==> Unable to find document in mongo with url: {}({})",
-                    url, err
+                    "Error attempting to find document: {}",
+                    err
                 )))
             }
         };
 
+        let last_hash: &String = &repo.last_hash;
+
         println!(
             "==> Comparing Repo({}) to MongoDB({:#?})",
             last_hash, document.last_hash
         );
 
-        return if *last_hash != document.last_hash.unwrap() {
+        if *last_hash != document.last_hash.unwrap() {
             let url = &document.id;
 
-            let clone_path = match self.github.clone_repo(url.as_str(), last_hash) {
-                Ok(clone_path) => {
-                    println!("==> {} Cloned Successfully", url);
-                    clone_path
+            let clone_path = self
+                .github
+                .clone_repo(url.as_str())
+                .map_err(|err| Error::GitHub(format!("Error attempting clone repo: {:#?}", err)))?;
+
+            let syft = Syft::new(clone_path.clone());
+
+            let mut syft_results: Vec<String> = vec![];
+
+            let mut total_build_targets = 0;
+
+            for (cataloger, build_targets) in cataloger_to_build_file_map {
```

Comment:
Looking at the error handling here, am I correct that if one of the build targets fails the whole repo will fail?  Is that what we want?  I truly don't know, but I'm wondering if it should try to be more fault tolerant like the out task loop.

Reply:
Made error handling fault tolerant